types/jasmine/index.d.ts

```diff
@@ -905,36 +907,28 @@ declare namespace jasmine {
         id: number;
         env: Env;
         description: string;
+        getFullName(): string;
     }
 
     interface Spec extends SuiteOrSpec {
-        getFullName(): string;
-        getResult(): any;
-        expect(actual: any): any;
-        execute(onComplete?: () => void, enabled?: boolean): any;
-        throwOnExpectationFailure: boolean;
-    }
+        new (attrs: any): any;
 
-    interface XSpec {
-        id: number;
+        setSpecProperty: typeof setSpecProperty;
+        getResult(): SpecResult;
+        execute(onComplete: (error?: Error) => void, excluded?: boolean, failSpecWithNoExpectations?: boolean): void;
+        pend(message?: string): void;
+        status(excluded?: boolean, failSpecWithNoExpectations?: boolean): string;
```

Comment:
@sgravrock These spec functions are not documented but I think they could be helpful to certain applications using Jasmine. Just wondering which ones (if any) I should add to the Spec type?

Reply:
at least `setSpecProperty` and `getResult` seem pretty useful.

Reply:
Please don't add any of those, or the `Suite` members below. None of them are safe to call at arbitrary times. `execute` is particularly scary -- I can't think of any way for user code to call it without corrupting Jasmine's internal state. But even something like `setSpecProperty` is only safe to call on the currently executing spec. The corresponding public APIs have safeguards that these internal methods don't.

One of the big lessons of Jasmine 1.x is that almost everything is a breaking change when user code has direct access to `Spec` and `Suite`. Hiding those objects was one of the main goals of the 2.0 rewrite. They were later re-exposed via `topSuite`, which was probably a mistake, but we can at least avoid making the problem worse by keeping the documented API as narrow as possible.

Thanks for asking.